### DISCUSSION TITLE
Add Uploads to a playlist

### DISCUSSION THREAD

**Discussion by LazyNinja435** [apastel/ytmusic-deleter#65] [Ideas]
Please consider including the feature to add the uploaded music to an existing or new playlist

**Comment by apastel** [comment]
Hmm I guess Github did not notify me that this was posted here, my bad.

Can you be more specific? This tool is mainly for deleting, not uploading, but I've made exceptions here and there. You want to add certain uploaded songs to an existing playlist? You can already do that in the YTM web app. Are you saying you want to add **all** of your uploads to one particular playlist? I could see that being useful only because it would take a long time to check the box for every upload, since they don't seem to let you do "Select All". Is that what you're asking for?

**Reply by LazyNinja435** [reply]
Yes for both. I'll check out the option to add certain songs to the existing playlist. Being able to add all uploaded songs to a playlist is a very helpful feature for me as I have 1000's of songs that I'm looking to upload

**Reply by apastel** [reply]
Another way to do it quicker is to find the uploaded album and add the album to the playlist, that at least lets you do it in bigger chunks.
![Peek 2024-10-16 14-19](https://github.com/user-attachments/assets/b8cfbaef-fde8-478d-b0b9-5c6f3d0a3e9a)

They do make it relatively easy to find all your uploads in one place, grouped by artist or by album or the full list of songs, but I can understand sometimes people would prefer a playlist.

**Reply by LazyNinja435** [reply]
Yes. That's how I did it for now. I had a simple Selenium script to check all my uploaded Albums so I could add them to the playlist. I still have a lot more songs to upload and I need to do the same on multiple accounts.

**Reply by apastel** [reply]
Selenium, nice.

Yeah I could see this being useful. Like a "Add All to Playlist" feature that would let you select either "Library" or "Uploads", and then it lets you select the playlist that you want to add all songs to. I'll add this idea to my feature requests, which currently this would be the one and only item on ;)

**Comment by apastel** [comment]
Ok so I went ahead and added a "Add All to Playlist" button. It lets you choose either "Uploads" or "Library" to add to a particular playlist. If you want to add it to a new playlist, just create a blank playlist first. The API adds the tracks in alphabetical order by trackname, so they aren't organized by artist or album, but you can use my "Sort Playlist" feature to sort it afterwards.

It's currently in pre-release here: https://github.com/apastel/ytmusic-deleter/releases/tag/2.7.0

Give it a try and let me know how it works out for you.

**Reply by LazyNinja435** [reply]
It worked perfectly. It took all of 2 seconds to add 1500 songs to a playlist. Thank you so much.
I've been trying the sort playlist option using different tools but none of them seem to be able to do it just by album title. Also, your app be sorting it based on Album Artist for my songs (Which could be an issue with my song's metadata).

**Reply by apastel** [reply]
Awesome, glad to hear it worked.

As far as the sorting goes, fortunately I have complete control over that function, since my tool sorts the tracks offline using my own criteria and then just issues "move" requests to move each song to the right place. Right now it sorts by Artist -> Album Title -> Track Title in that order. I have thought about expanding that to give people a choice of how they want to sort things. I can envision some kind of window where you have a bunch of attribute names like "Artist", "Album Title", "Track Title", "Track Number", "Duration", and then you can drag and drop those attributes into a little stack or something, so you can choose as many or few as you want, and you can determine the hierarchy. So in your case you would just take "Album Title" (and then maybe Track Number if you want). So that would give people complete control over how to sort things. Yeah, that would be sweet. Next feature to implement!

As far as the Album Artist thing goes, I'm not sure why that's happening for you. My tool is using the "artists" field for the specific track. The track objects I'm seeing do not even have a "albumArtist" attribute, even for ones that should, like a song from a movie soundtrack where the Album Artist is "Various Artists". Do you have any examples of songs in YTMusic that I can test out and replicate that problem?

**Reply by apastel** [reply]
You can sort based on just Album Title now in the latest version I just released: https://github.com/apastel/ytmusic-deleter/releases/tag/2.8.0
